diff -Naur 1.0-r0-orig/network@.service 1.0-r0/network@.service
--- 1.0-r0-orig/network@.service	2019-07-22 08:52:29.887900569 +0000
+++ 1.0-r0/network@.service	2019-07-22 08:52:42.039688137 +0000
@@ -4,7 +4,6 @@
 Type=oneshot
 RemainAfterExit=yes
 EnvironmentFile=/etc/device.properties
-ExecStartPre=/bin/sh -c "/lib/rdk/disableIpv6Autoconf.sh $MOCA_INTERFACE"
 ExecStartPre=/bin/sh -c 'if [ "$BUILD_TYPE" == "prod" ] && [ -f /proc/sys/kernel/sysrq ];then sysctl -w "kernel.sysrq=0"; fi'
 ExecStart=/sbin/ip link set dev %i up
 ExecStartPost=-/lib/rdk/disableUnusedInterfaces.sh %i
