diff --git a/systemd_units/meshwifi.service b/systemd_units/meshwifi.service
index 518ad5a..4c4fa27 100644
--- a/systemd_units/meshwifi.service
+++ b/systemd_units/meshwifi.service
@@ -22,7 +22,7 @@ ConditionPathExists=/tmp/wifi_initialized
 
 [Service]
 Type=oneshot
-WorkingDirectory=/usr/plume
+WorkingDirectory=/usr/opensync
 
 ExecStartPre=/bin/echo "Mesh Wifi Starting"
 ExecStart=/usr/ccsp/wifi/handle_mesh start
