diff --git a/source/MeshAgentSsp/cosa_mesh_apis.c b/source/MeshAgentSsp/cosa_mesh_apis.c
index 0a8d332..6b220f8 100644
--- a/source/MeshAgentSsp/cosa_mesh_apis.c
+++ b/source/MeshAgentSsp/cosa_mesh_apis.c
@@ -122,8 +122,8 @@ token_t sysevent_token_gs;
 token_t sysevent_token;
 static pthread_t sysevent_tid;

-const char urlOld[] = "NOC-URL-DEV";
-const char urlDefault[] = "NOC-URL-PROD";
+const char urlOld[] = "ssl:yellowstone.plume.comcast.net:443";
+const char urlDefault[] = "ssl:wildfire.plume.tech:443";
 const char meshServiceName[] = "meshwifi";
 const char meshDevFile[] = "/nvram/mesh-dev.flag";
 static bool gmssClamped = false;
