diff --git a/source/MeshAgentSsp/cosa_mesh_apis.c b/source/MeshAgentSsp/cosa_mesh_apis.c
index fd657d5..b4d441b 100644
--- a/source/MeshAgentSsp/cosa_mesh_apis.c
+++ b/source/MeshAgentSsp/cosa_mesh_apis.c
@@ -1470,7 +1470,7 @@ bool Mesh_GetEnabled(const char *name)
 
     return enabled;
 }
-
+#ifndef _PLATFORM_TURRIS_
 static void applyWifiRadioSettings(int radioId)
 {
     CCSP_MESSAGE_BUS_INFO *bus_info = (CCSP_MESSAGE_BUS_INFO *)bus_handle;
@@ -1506,6 +1506,7 @@ static void applyWifiRadioSettings(int radioId)
         bus_info->freefunc(faultParam);
     }
 }
+#endif
 
 void changeChBandwidth(int radioId, int channelBw) {
   CCSP_MESSAGE_BUS_INFO *bus_info = (CCSP_MESSAGE_BUS_INFO *)bus_handle;
@@ -1543,7 +1544,12 @@ void changeChBandwidth(int radioId, int channelBw) {
         bus_info->freefunc( faultParam );
     }
 
+/* On Turris omnia platform change bandwidth is already done with wifi_pushRadioChannel2 call.
+ * We want to copy config to datamodel, but we don't want to apply radio settings
+ */
+#ifndef _PLATFORM_TURRIS_
     applyWifiRadioSettings(radioId);
+#endif
 }

