diff --git a/src/journal/journal-send.c b/src/journal/journal-send.c
index 5e8a3e3200..f017a301ee 100644
--- a/src/journal/journal-send.c
+++ b/src/journal/journal-send.c
@@ -358,8 +359,8 @@ static int fill_iovec_perror_and_send(const char *message, int skip, struct iove
                 if (errno == 0) {
                         char error[sizeof("ERRNO=")-1 + DECIMAL_STR_MAX(int) + 1];
 
-                        if (j != buffer + 8 + k)
-                                memmove(buffer + 8 + k, j, strlen(j)+1);
+                        //if (j != buffer + 8 + k)
+                        //        memmove(buffer + 8 + k, j, strlen(j)+1);
 
                         memcpy(buffer, "MESSAGE=", 8);
 
diff --git a/src/libsystemd/sd-bus/bus-error.c b/src/libsystemd/sd-bus/bus-error.c
index 26219bdeed..3c895c061e 100644
--- a/src/libsystemd/sd-bus/bus-error.c
+++ b/src/libsystemd/sd-bus/bus-error.c
@@ -378,7 +378,7 @@ static void bus_error_strerror(sd_bus_error *e, int error) {
 
                 errno = 0;
                 x = strerror_r(error, m, k);
-                if (errno == ERANGE || strlen(x) >= k - 1) {
+                if (errno == ERANGE || strlen(m) >= k - 1) {
                         free(m);
                         k *= 2;
                         continue;
