diff --git a/src/dm/src/dm_manager.c b/src/dm/src/dm_manager.c
index 728e6f2..a3163dd 100644
--- a/src/dm/src/dm_manager.c
+++ b/src/dm/src/dm_manager.c
@@ -122,34 +122,12 @@ bool pid_dir(void)
 
 bool init_managers()
 {
-    int i;
-
     /* terminate all running managers */
     if (!pid_dir())
     {
         LOG(ERR, "Can't create PID folder: path=%s", CONFIG_DM_PID_PATH);
     }
 
-    /*
-     * Legacy code: Convert the TARGET API manager list to a dynamic
-     * list.
-     */
-    for (i = 0; i < (int)target_managers_num; i++)
-    {
-        struct dm_manager *dm = calloc(1, sizeof(*dm));
-
-        /*
-         * Add manager to global list of managers
-         */
-        LOG(INFO, "Adding legacy manager: %s", target_managers_config[i].name);
-
-        dm_manager_register(
-                target_managers_config[i].name,
-                target_managers_config[i].needs_plan_b,
-                target_managers_config[i].always_restart,
-                target_managers_config[i].restart_delay);
-    }
-
     if (!dm_manager_start_all())
     {
         LOG(ERR, "Failed to start at least one manager.");
