diff --git a/source/dmltad/diag.c b/source/dmltad/diag.c
index 23a5f3b..7fb0bc6 100644
--- a/source/dmltad/diag.c
+++ b/source/dmltad/diag.c
@@ -174,7 +174,7 @@ static int inet_is_local(const struct in_addr *addr)
      * 3. unicast to address in same network e.g,. "xxx.xxx.xxx.xxx/xx" 
      */
      snprintf(cmd, sizeof(cmd), "ip route show table local");
-#ifdef _PLATFORM_RASPBERRYPI_
+#if defined(_PLATFORM_RASPBERRYPI_) || (_PLATFORM_TURRIS_)
 /*
 rb mode is not supported to create pipe
 */
@@ -414,10 +414,11 @@ static void *diag_task(void *arg)
     pthread_mutex_lock(&diag->mutex);
     cfg = diag->cfg;
     pthread_mutex_unlock(&diag->mutex);
-#ifdef _PLATFORM_RASPBERRYPI_
+#if defined(_PLATFORM_RASPBERRYPI_) || (_PLATFORM_TURRIS_)
 /**
  inet_pton is failing because of extra quotes in ip address (cfg.host)
 **/
+    printf("Turris work dude\n");
     int len = strlen(cfg.host);
     if( (cfg.host[0] == '\'') && (cfg.host[len-1] == '\'') )
     {
diff --git a/source/dmltad/diag_ping.c b/source/dmltad/diag_ping.c
index 47341c6..7e7f58d 100644
--- a/source/dmltad/diag_ping.c
+++ b/source/dmltad/diag_ping.c
@@ -109,7 +109,7 @@ static diag_err_t ping_start(diag_obj_t *diag, const diag_cfg_t *cfg, diag_stat_
     else
         cnt = cfg->cnt;
 
-    left -= snprintf(cmd + strlen(cmd), left, "ping %s ", cfg->host);
+    left -= snprintf(cmd + strlen(cmd), left, "ping ");
     if (strlen(cfg->ifname))
         left -= snprintf(cmd + strlen(cmd), left, "-I %s ", cfg->ifname);
     if (cnt)
@@ -122,6 +122,7 @@ static diag_err_t ping_start(diag_obj_t *diag, const diag_cfg_t *cfg, diag_stat_
     if (cfg->tos)
         left -= snprintf(cmd + strlen(cmd), left, "-Q %u ", cfg->tos);
 #endif
+    left -= snprintf(cmd + strlen(cmd), left, "%s ", cfg->host);
     left -= snprintf(cmd + strlen(cmd), left, "2>&1 ");
 
     if (left <= strlen(awkcmd) + 1)

