diff --git a/source/platform/platform_hal.c b/source/platform/platform_hal.c
index b963113..8f31163 100644
--- a/source/platform/platform_hal.c
+++ b/source/platform/platform_hal.c
@@ -143,3 +143,29 @@ INT platform_hal_SetDeviceCodeImageValid(BOOLEAN flag)
 {
	return RETURN_OK;
 }
+
+INT platform_hal_getCMTSMac(CHAR *pValue)
+{
+     if (pValue == NULL)
+     {
+         return RETURN_ERR;
+     }
+    strcpy(pValue,"00:00:00:00:00:00");
+    return RETURN_OK;
+}
+
+/* platform_hal_SetSNMPOnboardRebootEnable() function */
+/**
+* @description : Set SNMP Onboard Reboot Enable value
+*                to allow or ignore SNMP reboot
+* @param IN    : pValue - SNMP Onboard Reboot Enable value
+                 ("disable" or "enable")
+*
+* @return      : The status of the operation
+* @retval      : RETURN_OK if successful
+* @retval      : RETURN_ERR if any error is detected
+*/
+INT platform_hal_SetSNMPOnboardRebootEnable(CHAR* pValue)
+{
+	return RETURN_OK;
+}
