diff --git a/custom/comcast/source/TR-181/custom_ml/cosa_deviceinfo_dml_custom.c b/custom/comcast/source/TR-181/custom_ml/cosa_deviceinfo_dml_custom.c
index 3bee024..e50720e 100644
--- a/custom/comcast/source/TR-181/custom_ml/cosa_deviceinfo_dml_custom.c
+++ b/custom/comcast/source/TR-181/custom_ml/cosa_deviceinfo_dml_custom.c
@@ -330,7 +330,7 @@ DeviceInfo_GetParamStringValue_Custom
 
 	if( AnscEqualString(ParamName, "X_COMCAST-COM_MTA_MAC", TRUE))
 	{
-#if !defined(_PLATFORM_RASPBERRYPI_)
+#if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)
   	   CosaDmlDiGetMTAMacAddress(NULL, pValue,pulSize);
 #endif
 	   return 0;
@@ -362,7 +362,7 @@ DeviceInfo_GetParamStringValue_Custom
 
 	if( AnscEqualString(ParamName, "X_COMCAST-COM_WAN_IPv6", TRUE))
 	{
-#if !defined(_PLATFORM_RASPBERRYPI_)
+#if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)
 	   CosaDmlDiGetRouterIPv6Address(NULL, pValue,pulSize);
            #if defined(_COSA_FOR_BCI_)
            if (syscfg_set(NULL, "wanIPv6Address",pValue) != 0) {
@@ -379,7 +379,7 @@ DeviceInfo_GetParamStringValue_Custom
 
 	if( AnscEqualString(ParamName, "X_COMCAST-COM_MTA_IP", TRUE))
 	{
-#if !defined(_PLATFORM_RASPBERRYPI_)
+#if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)
    	   CosaDmlDiGetMTAIPAddress(NULL, pValue,pulSize);
 #endif
 	   return 0;
-- 
2.16.2
--- git/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c	2019-08-20 10:17:03.754398442 +0000
+++ git2/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c	2019-08-20 10:22:28.782347638 +0000
@@ -341,7 +341,7 @@
     return 0;
 }
 
-#if defined(_PLATFORM_RASPBERRYPI_)
+#if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
 static int
 DmSetBool(const char *param, BOOL value)
 {
@@ -3767,7 +3767,7 @@
         }
 #endif
         
-#if defined(_PLATFORM_RASPBERRYPI_)
+#if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
 	char buf[7] = {0};
 	BOOL value;
 	snprintf(buf,sizeof(buf),"%ld",bridge_info.mode);

