Signed-off-by: kaviya.kumaresan@ltts.com
Subject: Including PLATFORM_TURRIS that supports the wan-manager integeration for Turris-Omnia

diff --git a/source/TR-181/board_sbapi/cosa_ethernet_apis.c b/source/TR-181/board_sbapi/cosa_ethernet_apis.c
index fd3d24b..dfa97b0 100644
--- a/source/TR-181/board_sbapi/cosa_ethernet_apis.c
+++ b/source/TR-181/board_sbapi/cosa_ethernet_apis.c
@@ -317,12 +317,14 @@ typedef enum WanMode
 #define ETHWAN_DEF_INTF_NAME "eth5"
 #elif defined (INTEL_PUMA7)
 #define ETHWAN_DEF_INTF_NAME "nsgmii0"
+#elif defined (_PLATFORM_TURRIS_)
+#define ETHWAN_DEF_INTF_NAME "eth2"
 #else
 #define ETHWAN_DEF_INTF_NAME "eth0"
 #endif
 
 
-#if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
+#if defined(_PLATFORM_RASPBERRYPI_) 
 #define ETHWAN_DEF_INTF_NAME "eth0"
 #endif
 
@@ -1601,7 +1603,7 @@ CosaDmlEthInit(
         }
     }
 #else
-    #if defined(_PLATFORM_RASPBERRYPI_)
+    #if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
     char wanPhyName[20] = {0},out_value[20] = {0};
 
     if (!syscfg_get(NULL, "wan_physical_ifname", out_value, sizeof(out_value)))
diff --git a/source/TR-181/board_sbapi/cosa_ethernet_manager.c b/source/TR-181/board_sbapi/cosa_ethernet_manager.c
index 3c70853..02e5b74 100644
--- a/source/TR-181/board_sbapi/cosa_ethernet_manager.c
+++ b/source/TR-181/board_sbapi/cosa_ethernet_manager.c
@@ -21,7 +21,7 @@
 /* ---- Include Files ---------------------------------------- */
 #include "cosa_ethernet_manager.h"
 #include "cosa_ethernet_apis.h"
-#if defined(_PLATFORM_RASPBERRYPI_)
+#if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
 #include "syscfg.h"
 #endif
 
@@ -329,7 +329,7 @@ static ethSmState_t Transition_EthWanLinkFound(PETH_SM_PRIVATE_INFO pstInfo)
 #if defined(FEATURE_RDKB_WAN_AGENT)
     if (ANSC_STATUS_SUCCESS != CosaDmlEthCreateEthLink(pstInfo->Name, stGlobalInfo.Path))
 #elif defined(FEATURE_RDKB_WAN_MANAGER)
-    #if defined(_PLATFORM_RASPBERRYPI_)     
+    #if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)     
     CHAR wanPhyName[20] = {0},out_value[20] = {0};
     if (!syscfg_get(NULL, "wan_physical_ifname", out_value, sizeof(out_value)))
     {
