diff --git a/source/lm/lm_wrapper.c b/source/lm/lm_wrapper.c
index 8a0dd48..ee153df 100644
--- a/source/lm/lm_wrapper.c
+++ b/source/lm/lm_wrapper.c
@@ -1081,7 +1081,7 @@ int lm_wrapper_get_arp_entries (char netName[LM_NETWORK_NAME_SIZE], int *pCount,
     char            out[32]         = {0};

 	// XB6/XF3 Do not have this interface. Remove constant warnings.
-#if !defined(INTEL_PUMA7) && !defined(_COSA_BCM_MIPS_) && !defined(_COSA_BCM_ARM_)
+#if !defined(INTEL_PUMA7) && !defined(_COSA_BCM_MIPS_) && !defined(_COSA_BCM_ARM_) && !defined(_PLATFORM_TURRIS_)
      if(pAtomBRMac[0] == '\0' || pAtomBRMac[0] == ' ') {
 		_ansc_sprintf(cmd, "ifconfig l2sd0 | grep HWaddr | awk '{print $5}' | cut -c 1-14\n" );
 		_get_shell_output(cmd, out, sizeof(out));
