diff --git a/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c b/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
index 475b014..7d7c144 100644
--- a/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
@@ -107,6 +107,10 @@
 #include "ccsp_vendor.h"
 #endif
 
+#ifdef _PLATFORM_TURRIS_
+#include "ccsp_vendor.h"
+#endif
+
 #ifdef _COSA_SIM_
 
 // this file is in integration_src.intel_usg_arm directory
diff --git a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
index e1195b0..dc8ff04 100755
--- a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
@@ -307,6 +307,9 @@ static int _is_in_linux_bridge(char * if_name, char * br_name)
 #ifdef _COSA_BCM_MIPS_
     #define COSA_USG_IF_NUM 3 // we only have 3 interfaces for XF3
 #endif
+#ifdef _PLATFORM_TURRIS_
+    #define COSA_USG_IF_NUM 3
+#endif
 
 typedef struct USG_IF_CFG
 {
@@ -321,8 +321,10 @@ USG_IF_CFG_T g_usg_if_cfg[COSA_USG_IF_NUM] =
     {"erouter0",    COSA_DML_LINK_TYPE_EthLink, TRUE},
 #ifdef _COSA_INTEL_USG_ARM_
 #ifndef _PLATFORM_RASPBERRYPI_
+#ifndef _PLATFORM_TURRIS_
     {"wan0",        COSA_DML_LINK_TYPE_DOCSIS,  TRUE},  /*DH  wan0 should never appear here -- CM extensions are for DOCSIS interfaces */
 #endif
+#endif
 #endif
     {"lo",          COSA_DML_LINK_TYPE_EthLink, FALSE}, /*DH  change the value of gDmsbIpIfLoopbackInstNum too, if "lo" is moved to a different location */
     /*
diff --git a/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c b/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c
index ce95e78..1d78c3e 100644
--- a/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_x_cisco_com_devicecontrol_apis.c
@@ -3406,7 +3406,7 @@ CosaDmlLanMngm_SetConf(ULONG ins, PCOSA_DML_LAN_MANAGEMENT pLanMngm)
     char str[IFNAME_SZ];    
     char *enableStr;
     napt_mode_t napt;
-#if !defined(_CBR_PRODUCT_REQ_) && !defined(_PLATFORM_RASPBERRYPI_) // MOCA is not present for TCCBR environment and RaspberryPi environment
+#if !defined(_CBR_PRODUCT_REQ_) && !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_) // MOCA is not present for TCCBR environment and RaspberryPi environment
 	parameterValStruct_t **valMoCAstatus;
 	char pMoCAComponentName[64]="eRT.com.cisco.spvtg.ccsp.moca";
 	char pComponentPath[64]="/com/cisco/spvtg/ccsp/moca";
@@ -3425,7 +3425,7 @@ CosaDmlLanMngm_SetConf(ULONG ins, PCOSA_DML_LAN_MANAGEMENT pLanMngm)
         Utopia_SetLanMngmInsNum(&utctx, pLanMngm->InstanceNumber);
         bridge_info.mode = pLanMngm->LanMode == COSA_DML_LanMode_BridgeStatic ? BRIDGE_MODE_STATIC : BRIDGE_MODE_OFF; 
         Utopia_SetBridgeSettings(&utctx,&bridge_info);
-#if !defined(_CBR_PRODUCT_REQ_) && !defined(_PLATFORM_RASPBERRYPI_) // MOCA is not present for TCCBR environment and RaspberryPi environment
+#if !defined(_CBR_PRODUCT_REQ_) && !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_) // MOCA is not present for TCCBR environment and RaspberryPi environment
 		ret = CcspBaseIf_getParameterValues(
 		    bus_handle,
 		    pMoCAComponentName,
diff --git a/source/TR-181/middle_layer_src/cosa_x_cisco_com_devicecontrol_dml.c b/source/TR-181/middle_layer_src/cosa_x_cisco_com_devicecontrol_dml.c
index 5563c26..49f0ebe 100644
--- a/source/TR-181/middle_layer_src/cosa_x_cisco_com_devicecontrol_dml.c
+++ b/source/TR-181/middle_layer_src/cosa_x_cisco_com_devicecontrol_dml.c
@@ -2020,7 +2020,7 @@ LanMngm_Validate
     /* check subnetmask */
     /* Subnet mask MUST accept ONLY the following IP addresses: */
     /* 255.255.255.0, 255.255.0.0, 255.0.0.0, 255.255.255.128, 255.255.255.252 */
-#if defined(_XB6_PRODUCT_REQ_)  || defined(_PLATFORM_RASPBERRYPI_)
+#if defined(_XB6_PRODUCT_REQ_)  || defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
     if(pLanMngm->LanSubnetMask.Value != 0x00FFFFFF &&
        pLanMngm->LanSubnetMask.Value != 0x0000FFFF &&
        pLanMngm->LanSubnetMask.Value != 0x000000FF &&  
@@ -2099,7 +2099,7 @@ LanMngm_Validate
     }else if(pLanMngm->LanIPAddress.Dot[0] == 172 && pLanMngm->LanIPAddress.Dot[1] >= 16 && pLanMngm->LanIPAddress.Dot[1] <= 31){
         return TRUE;
     }
-#if defined(_XB6_PRODUCT_REQ_) || defined (_CBR_PRODUCT_REQ_) || defined(_PLATFORM_RASPBERRYPI_)
+#if defined(_XB6_PRODUCT_REQ_) || defined (_CBR_PRODUCT_REQ_) || defined(_PLATFORM_RASPBERRYPI_) ||  defined(_PLATFORM_TURRIS)
     else if((pLanMngm->LanIPAddress.Value & 0x0000FFFF) == 0x0000A8C0)
 #else
    else if((pLanMngm->LanIPAddress.Value & 0xFFFF0000) == 0xC0A80000)
