diff --git a/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c b/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
index 079e9571..a7a2b59d 100755
--- a/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_dhcpv6_apis.c
@@ -7931,12 +7931,6 @@ void UnSetV6RouteFromTable(char* ifname , char* route_addr,int metric_val, int t
 
 #endif
 
-#ifdef WAN_FAILOVER_SUPPORTED
-
-#define MESH_WAN_IFNAME "dmsb.Mesh.WAN.Interface.Name"
-#define MESH_WAN_WAN_IPV6ADDR "MeshWANInterface_UlaAddr"
-#define MESH_REMWAN_WAN_IPV6ADDR "MeshRemoteWANInterface_UlaAddr"
-
 int Get_Device_Mode()
 {
     int deviceMode = 0;
@@ -7951,6 +7945,12 @@ int Get_Device_Mode()
 
 }
 
+#ifdef WAN_FAILOVER_SUPPORTED
+
+#define MESH_WAN_IFNAME "dmsb.Mesh.WAN.Interface.Name"
+#define MESH_WAN_WAN_IPV6ADDR "MeshWANInterface_UlaAddr"
+#define MESH_REMWAN_WAN_IPV6ADDR "MeshRemoteWANInterface_UlaAddr"
+
 void getMeshWanIfName(char *mesh_wan_ifname,int size)
 {
     char* val=NULL;
