diff --git a/source/TR-181/middle_layer_src/plugin_main_apis.c b/source/TR-181/middle_layer_src/plugin_main_apis.c
index e9f2b5e..13ba6bc 100644
--- a/source/TR-181/middle_layer_src/plugin_main_apis.c
+++ b/source/TR-181/middle_layer_src/plugin_main_apis.c
@@ -292,7 +292,7 @@ CosaBackEndManagerInitialize
     returnStatus = CosaDmlMlanInit((ANSC_HANDLE)pMyObject, &pMyObject->hMultiLan);
     AnscTraceWarning(("  CosaDmlMlanInit -- status %d!\n", returnStatus));
 
-#if !defined(INTEL_PUMA7) && !defined(_COSA_BCM_MIPS_) && !defined(_COSA_BCM_ARM_)
+#if !defined(INTEL_PUMA7) && !defined(_COSA_BCM_MIPS_) && !defined(_COSA_BCM_ARM_) && !defined(_PLATFORM_TURRIS_)
     printf("pnm-status is renamed to bring-lan and set in PSM for XB3\n");
 #else
    printf("**************** sysevent set pnm-status up \n");
