Signed-off-by: kaviya.kumaresan@ltts.com
Subject: To avoid gSsidCount undeclared error while compiling Wifi_hal3.0

diff --git a/source/TR-181/sbapi/cosa_wifi_apis.c b/source/TR-181/sbapi/cosa_wifi_apis.c
index 37019bc..aba7811 100644
--- a/source/TR-181/sbapi/cosa_wifi_apis.c
+++ b/source/TR-181/sbapi/cosa_wifi_apis.c
@@ -9552,6 +9552,7 @@ CosaDmlWiFiFactoryReset
                 }
             }
         }
+#ifndef WIFI_HAL_VERSION_3
 #if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
 	// Reset Band Steering parameters
 	int bsIndex = 0;
@@ -9560,6 +9561,7 @@ CosaDmlWiFiFactoryReset
 	    CosaDmlWiFiGetBSFactoryResetPsmData(bsIndex, bsIndex+1);
 	}
 #endif
+#endif
     } else
     {
         // Only Apply to FactoryResetSSID list
