diff --git a/scripts/handle_mesh b/scripts/handle_mesh
index 5031842..7cc6eee 100644
--- a/scripts/handle_mesh
+++ b/scripts/handle_mesh
@@ -57,11 +57,19 @@ is_opensync_name()
     return 1
 }
 
+is_opensync_service()
+{
+    systemctl list-unit-files opensync.service | grep -q opensync.service
+}
+
 if [ $# -eq 0 ]; then
  echo "No arguments passed"
  exit 0
 else
  platform_checks
+ if is_opensync_service; then
+        systemctl $1 opensync.service
+ else
  if [ "$OPENSYNC_ENABLE" == "true" ];then
     if [ -z "$(pidof ovsdb-server)" ]
     then
@@ -71,9 +79,10 @@ else
         /usr/opensync/scripts/managers.init $1
     else
         echo "Opensync will be effective only after reboot"
-        /usr/plume/scripts/managers.init $1
+        /usr/opensync/scripts/managers.init $1
     fi
  else
-  /usr/plume/scripts/managers.init $1
+  /usr/opensync/scripts/managers.init $1
+ fi
  fi
 fi

