From b057b158cbeac3a70801f898eb02c05162bc4095 Mon Sep 17 00:00:00 2001
From: rnarayanan <Rajkumar.Narayanan@lnttechservices.com>
Date: Wed, 24 Apr 2019 09:35:52 +0000
Subject: [PATCH] add ccsp_vendor header file

Signed-off-by: rnarayanan <Rajkumar.Narayanan@lnttechservices.com>
---
 ccsp_vendor.h | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 ccsp_vendor.h

diff --git a/ccsp_vendor.h b/ccsp_vendor.h
new file mode 100644
index 00000000..1f0d7715
--- /dev/null
+++ b/ccsp_vendor.h
@@ -0,0 +1,4 @@
+#define CONFIG_VENDOR_NAME "Turris CZ"
+#define CONFIG_VENDOR_ID "0xFFFFFFFE"
+#define CONFIG_TI_GW_DESCRIPTION "Turris Omnia Gateway device"
+
-- 
2.19.0

