diff --git a/systemd_units/PsmSsp.service b/systemd_units/PsmSsp.service
index cb4d530d..648a55f9 100644
--- a/systemd_units/PsmSsp.service
+++ b/systemd_units/PsmSsp.service
@@ -29,7 +29,6 @@ Environment="Subsys=eRT."
 Environment="LOG4C_RCPATH=/etc"
 EnvironmentFile=/etc/device.properties
 ExecStartPre=/bin/sh -c '(/usr/ccsp/utopiaInitCheck.sh)'
-ExecStartPre=/bin/sh -c '(/usr/ccsp/log_psm.db.sh)'
 ExecStart=/usr/bin/PsmSsp -subsys $Subsys
 ExecStop=/bin/sh -c 'echo "`date`: Stopping/Restarting PsmSsp" >> ${PROCESS_RESTART_LOG}'
 Restart=always
