diff --git a/source/syscfg/lib/Makefile.am b/source/syscfg/lib/Makefile.am
index 50d14ea2..8bfc3789 100644
--- a/source/syscfg/lib/Makefile.am
+++ b/source/syscfg/lib/Makefile.am
@@ -20,7 +20,7 @@ AM_CFLAGS = -D_ANSC_LINUX
 AM_CFLAGS += -D_ANSC_USER
 AM_CFLAGS += -D_ANSC_LITTLE_ENDIAN_
 AM_CFLAGS += -D_GNU_SOURCE
-AM_CFLAGS += -DSC_POSIX_SEM
+AM_CFLAGS += -DSC_SYSV_SEM
 
 ACLOCAL_AMFLAGS = -I m4
 
