From 625c988c0bbc67d392a1a53e39c5b13b4db37c75 Mon Sep 17 00:00:00 2001
From: Mateusz Bajorski <mbajorski@plume.com>
Date: Thu, 11 Jul 2019 11:49:39 +0000
Subject: [PATCH 2/2] DIRTY HACK: Fix syntax error around __GNUC_PREREQ

---
 src/lib/const/inc/const.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/lib/const/inc/const.h b/src/lib/const/inc/const.h
index cef6be4..8bb9962 100644
--- a/src/lib/const/inc/const.h
+++ b/src/lib/const/inc/const.h
@@ -257,11 +257,11 @@ extern bool             _c_get_param_by_key(c_item_t *list, int list_sz, int key
  * Static assertions: The macro below causes a compile time error if COND is false.
  */
 
-#if (defined(__GNUC__) && __GNUC_PREREQ(4,7)) || defined(__clang__)
+//#if (defined(__GNUC__) && __GNUC_PREREQ(4,7)) || defined(__clang__)
 
-#define C_STATIC_ASSERT(COND,MSG) _Static_assert((COND), MSG)
+//#define C_STATIC_ASSERT(COND,MSG) _Static_assert((COND), MSG)
 
-#else
+//#else
 
 /*
  * XXX: The code below produces a warning with -Wall, make sure to add -Wno-unused-local-typedefs when using
@@ -274,13 +274,13 @@ extern bool             _c_get_param_by_key(c_item_t *list, int list_sz, int key
 
 #define C_STATIC_ASSERT(COND,MSG) typedef char __LINE1(___STATIC_ASSERT,__LINE__)[(COND)?1:-1]
 
-#endif
+//#endif
 
 // Handle error: this statement may fall through [-Werror=implicit-fallthrough=]
-#if (defined(__GNUC__) && __GNUC_PREREQ(7,0))
-#define FALLTHROUGH __attribute__((fallthrough));
-#else
+//#if (defined(__GNUC__) && __GNUC_PREREQ(7,0))
+//#define FALLTHROUGH __attribute__((fallthrough));
+//#else
 #define FALLTHROUGH
-#endif
+//#endif
 
 #endif /* CONST_H_INCLUDED */
-- 
1.9.1

