diff --git a/build/vendor-arch.mk b/build/vendor-arch.mk
index 73fce2d..b7999f3 100644
--- a/build/vendor-arch.mk
+++ b/build/vendor-arch.mk
@@ -48,7 +48,17 @@ RDK_MODEL = omnia
 #RDK_CFLAGS  +=
 
 else
-$(error Unsupported RDK_MACHINE ($(RDK_MACHINE)).)
+  ifeq ($(RDK_MACHINE),turris-pod)
+
+  # TODO: Set correct OEM and model names
+  RDK_OEM = turris
+  RDK_MODEL = omnia
+
+  # TODO: Specify additional CFLAGS if needed
+  RDK_CFLAGS  += -DTURRIS_POD
+  else
+  $(error Unsupported RDK_MACHINE ($(RDK_MACHINE)).)
+  endif
 endif
 
 
