--- a/src/sm/src/sm_survey_report.c	2020-05-20 12:14:48.587156380 +0000
+++ src/sm/src/sm_survey_report.c	2020-05-20 13:59:48.377805976 +0000
@@ -968,7 +968,7 @@
     survey_ctx->threshold_record = *survey_entry;
 
     int threshold_util = result_entry.chan_tx + result_entry.chan_self;
-    LOGD("Checking %s %s survey threshold util: %d/%d delay: %ld/%d count: %d/%d",
+    LOGD("Checking %s %s survey threshold util: %d/%d delay: %lld/%d count: %d/%d",
          radio_get_name_from_cfg(radio_cfg_ctx),
          radio_get_scan_name_from_type(scan_type),
          threshold_util, request_ctx->threshold_util,
@@ -1100,7 +1100,7 @@
             scan_interval = SURVEY_MIN_SCAN_INTERVAL;
 
             // always scan if max_delay exceeded
-            LOGI("Force processing %s %s survey delay: %ld/%d count: %d/%d",
+            LOGI("Force processing %s %s survey delay: %lld/%d count: %d/%d",
                  radio_get_name_from_cfg(radio_cfg_ctx),
                  radio_get_scan_name_from_type(scan_type),
                  survey_ctx->threshold_time_delta, request_ctx->threshold_max_delay,
