Index: core/src/lib/inet/src/linux/inet_gretap.c
===================================================================
--- core.orig/src/lib/inet/src/linux/inet_gretap.c	2020-09-03 11:47:06.793454885 +0000
+++ core/src/lib/inet/src/linux/inet_gretap.c	2020-09-03 12:27:25.439372685 +0000
@@ -199,7 +199,7 @@
         snprintf(sremote_addr, sizeof(sremote_addr), PRI_osn_ip_addr, FMT_osn_ip_addr(self->in_remote_addr));
 
         if (osync_hal_inet_create_gre(self->inet.in_ifname, slocal_addr, sremote_addr,
-                    target_map_ifname_to_gre_bridge(self->in_ifparent), 1) != OSYNC_HAL_SUCCESS)
+                    self->in_ifparent, 1) != OSYNC_HAL_SUCCESS)
         {
             LOG(ERR, "inet_gretap: %s: cannot create GRE interface", self->inet.in_ifname);
             return false;
