diff --git a/src/bm/src/bm_events.c b/src/bm/src/bm_events.c
index 19bcd47..b2000cb 100644
--- a/src/bm/src/bm_events.c
+++ b/src/bm/src/bm_events.c
@@ -315,20 +315,20 @@ bm_events_handle_event(bsal_event_t *event)
         }
 
         /* This one managed by cloud */
+	/*
         if (abs(stats->probe.last_snr - event->data.probe_req.rssi) < client->preq_snr_thr) {
             LOGD("[%s] %s: %s skip preq report (%d %d)", bandstr, ifname, client->mac_addr,
                  stats->probe.last_snr, event->data.probe_req.rssi);
             break;
-        }
+        }*/
 
-        /* This is short time probe report filtering/limitation */
-        if (abs(stats->probe.last_snr - event->data.probe_req.rssi) <= BM_CLIENT_PREQ_SNR_TH &&
-            (now - last_probe) < BM_CLIENT_PREQ_TIME_TH) {
+    //   This is short time probe report filtering/limitation 
+        if ( (now - last_probe) < BM_CLIENT_PREQ_TIME_TH) {
             LOGD("[%s] %s: %s skip preq report (%d %d) short time", bandstr, ifname, client->mac_addr,
                  stats->probe.last_snr, event->data.probe_req.rssi);
             break;
         }
-
+   
         stats->probe.last_snr = event->data.probe_req.rssi;
         bm_stats_add_event_to_report( client, event, PROBE, false );
         break;
@@ -788,6 +788,7 @@ bm_events_handle_rssi_xing_cs(bm_client_t *client, bsal_event_t *event)
 static void
 bm_events_handle_rssi_xing(bm_client_t *client, bsal_event_t *event)
 {
+	/*
     if (event->data.rssi_change.low_xing == client->xing_low &&
         event->data.rssi_change.high_xing == client->xing_high &&
         (abs(event->data.rssi_change.rssi - client->xing_snr) <= BM_CLIENT_SNR_XING_DIFF)) {
@@ -795,7 +796,7 @@ bm_events_handle_rssi_xing(bm_client_t *client, bsal_event_t *event)
              event->data.rssi_change.rssi, client->xing_snr);
         return;
     }
-
+*/
     switch (client->steering_state) {
     case BM_CLIENT_CLIENT_STEERING:
         bm_events_handle_rssi_xing_cs(client, event);
