--- tools/lib/api/fs/tracing_path.c	2019-03-11 13:57:23.676631565 +0000
+++ tools1/lib/api/fs/tracing_path.c	2019-03-12 05:58:41.905325050 +0000
@@ -133,7 +133,9 @@
 	}
 		break;
 	default:
-		snprintf(buf, size, "%s", str_error_r(err, sbuf, sizeof(sbuf)));
+		//snprintf(buf, size, "%s", str_error_r(err, sbuf, sizeof(sbuf)));
+                strerror_r(err, sbuf, sizeof(sbuf));
+                snprintf(buf, size, "%s", sbuf);
 		break;
 	}
 
